app/src/main/java/com/asfoundation/wallet/ui/iab/payments/carrier/verify/CarrierVerifyPresenter.kt

```diff
@@ -67,6 +69,19 @@ class CarrierVerifyPresenter(
     )
   }
 
+  private fun handleChangeButtonClick() {
+    disposables.add(
+        view.changeButtonClick()
+            .observeOn(viewScheduler)
+            .doOnNext { _ ->
+              interactor.forgetPhoneNumber()
+              view.setSavedPhoneNumberVisibility(null, true)
```

Comment:
Maybe we can have two different methods
One to use saved phone number and other to clear.